### DISCUSSION TITLE
Lock file only updates for Bundler

### DISCUSSION THREAD

**Discussion by moetayuko** [renovatebot/renovate#11308] [Request Help]
### What would you like Renovate to be able to do?

I have a ruby project with Renovate bot enabled but not receiving any updates. After searching I found a similar issue fired two years ago, and a Renovate bot dev replied that `Gemfile.lock`-only update was not implemented at that time: https://github.com/renovatebot/renovate/issues/3860#issuecomment-497935910 which is exactly what I need. However, that issue was closed by the resolution of another bug found by the guy.

May I know the status of the feature? Is it implemented now?

### If you have any ideas on how this should be implemented, please tell us here.

Shouldn’t be too much work according to https://github.com/renovatebot/renovate/issues/3860#issuecomment-497938239

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
Renovate doesn't make any lockfile only updates by default. This is by design.

you need to configure `rangeStrategy=update-lockfile` or enable `lockFileMaintenance`.

https://docs.renovatebot.com/configuration-options/#rangestrategy
https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

Did you tried anything of this?

**Reply by moetayuko** [reply]
Thanks for your reply. TBH I was not aware of these options. Enabling `lockFileMaintenance` worked for me while `rangeStrategy=update-lockfile` didn't.

`update-lockfile` docs write "Update the lock file when in-range updates are available". However, I don't specify package versions or ranges in `Gemfile`, is it expected not to receive updates in my case?

**Reply by rarkins** [reply]
Currently update-lockfile only works if you have a constraint in Gemfile, however this is something we could improve so that it works even if the dependency has no constraint.

**Reply by moetayuko** [reply]
Gotcha. `lockFileMaintenance` is what I need so far.
